pyro/infer/reparam/neutra.py

```diff
@@ -71,26 +84,31 @@ def __call__(self, name, fn, obs):
                                  "`get_transform` method that does not depend on the "
                                  "model's `*args, **kwargs`") from e
 
-            z_unconstrained = pyro.sample("{}_shared_latent".format(name),
-                                          self.guide.get_base_dist().mask(False))
+            with ExitStack() as stack:
+                for plate in self.guide.plates.values():
+                    stack.enter_context(block_plate(dim=plate.dim, strict=False))
```

Comment:
I just wonder why we need to `block_plate` here (and some other places). Could you elaborate a bit?

Reply:
The problem addressed is with in models like
```py
@neutra.reparam()
def model():
    with pyro.plate("plate", 10):
        pyro.sample("x", dist.Normal(0,1))
```
where the `z_unconconstrained` shared latent is a single latent variable that is not independent along batch dimensions; rather it should have empty `batch_shape` (except for particle plates that are outside the guide and hence not blocked) and has `event_shape = (fn.shape().numel(),)`.  Without this `block` statement the `z_unconstrained` would have `event_shape = (10,)` but would then be broadcast up by an enclosing plate to `batch_shape + event_shape = (10, 10)`.  This this `block_plate` merely hoists that shared latent variable outside of all plates.

I believe this is what we want to do with `NeuTra`, but IIUC it prevents "batched neutra" in that we won't be able to share a single normalizing flow along a batch dimension and subsample.  I'm not sure whether that prevented use case is realistic though, since our HMC and NUTS do not support data subsampling.